docs/tutorials/setting-up-python-pycharm.md

```diff
@@ -1,83 +1,71 @@
+## Log in to ev3dev
 
-[network]: /docs/tutorials
-
-## Log in to ev3dev ##
-
-Now you can either attach a screen and keyboard to it for the setup, or
-connect to it via ssh. I prefer the latter. My router has a DNS that registers
-hostnames. It allows me to start a Terminal (on Mac) or Putty (on Windows) and
-simply do:
-
-`ssh robot@ev3dev`
+Now you can either or connect your ev3dev device [via ssh][ssh] or attach a screen and keyboard to it
+if using RPi. To log in via ssh, start a Terminal (on Mac) or Putty (on Windows) and type:
 
-If you don't have a DNS try: `robot@ev3dev.local` or connect to it's IP
-address directly. eg. `robot@192.168.0.2`
+`ssh robot@ev3dev.local`
 
-The password is `maker` by default. If you used screen and keyboard, you're
-automatically logged in.
+or use the IP address. eg. `robot@192.168.0.2`
 
-##Configure the right BrickPi##
+The password is `maker` by default.
 
-You're logged in to the command line. Cool! This step will activate the
-BrickPi. Skip it if you have a normal EV3. Now several Mindstorms shields
-are available for the RPi, we have to setup the right one first. That is,
-if you haven't done so already after burning the image. For this go:
+{% include icon.html type="info" %}
+If you are not using the EV3, but instead using an RPi solution there are
+several Mindstorms shields available now. You have to setup the right one.
+To do that type.
 
-`sudo nano /boot/flash/config.txt` (the password is still `maker`)
-Scroll down and uncomment the appropriate lines for your hardware.
+`sudo nano /boot/flash/config.txt` (the password is still `maker`)<br>
+Scroll down and uncomment the appropriate lines for your hardware.<br>
+Finally do `sudo reboot`.
+{:.alert .alert-info}
 
 {% include icon.html type="info" %}
 I have multiple Raspberry Pis and Mindstorms brains. I like to keep
 them apart on the network by giving them separate names. I renamed mine like
 this:<br />`sudo hostnamectl set-hostname ev3dev-rpi`
 {:.alert .alert-info}
 
-Finally do `sudo reboot`.
 
-## Set up development environment ##
 
+## Set up development environment
+
+{% include icon.html type="danger" %}
+The first half of this tutorial assumes you are using the free version of PyCharm.
+However, if you have access to the Professional Version of PyCharm, which is free for students
+and some Open Source projects, you should probably read the PyCharm Professional version
+alternatives shown later.
+{:.alert .alert-danger}
+
+{% include icon.html type="info" %}
 I prefer to have a passwordless ssh to the robot. For this I [add a user with
 the same login name as on my development computer.][users] Remember to give
 yourself sudo rights as explained on that same page. After adding that user,
-you can [set up passwordless ssh][passwordless].
+you can [set up passwordless ssh][passwordless].<br>
+With passwordless ssh and the same username on the ev3dev machine, you can go
+`ssh ev3dev` and you're in!
```

Comment:
Again, we have the problem of which hostname to use. This will work on Windows (except on Windows, you will probably be using PuTTY) and if you are using WiFi and your router has a name server that picks up names from DHCP requests. In other words, this command will not work for most people.